src/GitHub.Api/Git/Repository.cs

```diff
@@ -64,64 +58,69 @@ private void CacheContainer_OnCacheInvalidated(CacheType cacheType)
             }
         }
 
-        private void FireGitLogCacheUpdated(DateTimeOffset dateTimeOffset)
+        private void HandleRepositoryInfoCacheUpdatedEvent(CacheUpdateEvent cacheUpdateEvent)
         {
-            Logger.Trace("GitLogCacheUpdated {0}", dateTimeOffset);
-            GitLogCacheUpdated?.Invoke(new CacheUpdateEvent { UpdatedTimeString = dateTimeOffset.ToString() });
+            Logger.Trace("RepositoryInfoCache Updated {0}", cacheUpdateEvent.UpdatedTimeString);
+            CurrentBranchChanged?.Invoke(cacheUpdateEvent);
+            CurrentRemoteChanged?.Invoke(cacheUpdateEvent);
+            CurrentBranchAndRemoteChanged?.Invoke(cacheUpdateEvent);
         }
 
-        private void FireBranchCacheUpdated(DateTimeOffset dateTimeOffset)
+        private void HandleGitLocksCacheUpdatedEvent(CacheUpdateEvent cacheUpdateEvent)
         {
-            Logger.Trace("BranchCacheUpdated {0}", dateTimeOffset);
-            BranchCacheUpdated?.Invoke(new CacheUpdateEvent { UpdatedTimeString = dateTimeOffset.ToString() });
+            Logger.Trace("GitLocksCache Updated {0}", cacheUpdateEvent.UpdatedTimeString);
+            LocksChanged?.Invoke(cacheUpdateEvent);
         }
 
-        private void FireGitStatusCacheUpdated(DateTimeOffset dateTimeOffset)
+        private void HandleGitStatucCacheUpdatedEvent(CacheUpdateEvent cacheUpdateEvent)
         {
-            Logger.Trace("GitStatusCacheUpdated {0}", dateTimeOffset);
-            GitStatusCacheUpdated?.Invoke(new CacheUpdateEvent { UpdatedTimeString = dateTimeOffset.ToString() });
+            Logger.Trace("GitStatusCache Updated {0}", cacheUpdateEvent.UpdatedTimeString);
+            StatusChanged?.Invoke(cacheUpdateEvent);
         }
 
-        private void FireGitLocksCacheUpdated(DateTimeOffset dateTimeOffset)
+        private void HandleGitLogCacheUpdatedEvent(CacheUpdateEvent cacheUpdateEvent)
         {
-            Logger.Trace("GitStatusCacheUpdated {0}", dateTimeOffset);
-            GitLockCacheUpdated?.Invoke(new CacheUpdateEvent { UpdatedTimeString = dateTimeOffset.ToString() });
+            Logger.Trace("GitLogCache Updated {0}", cacheUpdateEvent.UpdatedTimeString);
+            LogChanged?.Invoke(cacheUpdateEvent);
         }
 
-        private void FireRepositoryInfoCacheUpdated(DateTimeOffset dateTimeOffset)
+        private void HandleBranchCacheUpdatedEvent(CacheUpdateEvent cacheUpdateEvent)
         {
-            Logger.Trace("RepositoryInfoCacheUpdated {0}", dateTimeOffset);
-            RepositoryInfoCacheUpdated?.Invoke(new CacheUpdateEvent { UpdatedTimeString = dateTimeOffset.ToString() });
+            Logger.Trace("BranchCache Updated {0}", cacheUpdateEvent.UpdatedTimeString);
+            LocalBranchListChanged?.Invoke(cacheUpdateEvent);
+            RemoteBranchListChanged?.Invoke(cacheUpdateEvent);
+            LocalAndRemoteBranchListChanged?.Invoke(cacheUpdateEvent);
```

Comment:
Same here, do we need this? These events are raised from the main thread so whoever needs to listen to both events can subscribe to both and get signaled before anything else happens (like OnUI or other code), so having a combined event feels redundant.